docker-bake.hcl

```diff
@@ -0,0 +1,272 @@
+variable "GIT_REVISION" {
+  default = "unknown"
+}
+
+variable "VERSION" {
+  default = "latest"
+}
+
+variable "GITHUB_REF_NAME" {}
+
+variable "GITHUB_BASE_REF" {}
+
+group "default" {
+  targets = ["janus", "interop_binaries"]
+}
+
+group "janus" {
+  targets = [
+    "janus_aggregator",
+    "janus_aggregation_job_creator",
+    "janus_aggregation_job_driver",
+    "janus_collection_job_driver",
+    "janus_cli",
+  ]
+}
+
+group "interop_binaries" {
+  targets = [
+    "janus_interop_client",
+    "janus_interop_aggregator",
+    "janus_interop_collector",
+  ]
+}
+
+group "release" {
+  targets = ["janus_release", "interop_binaries_release"]
+}
+
+group "janus_release" {
+  targets = [
+    "janus_aggregator_release",
+    "janus_aggregation_job_creator_release",
+    "janus_aggregation_job_driver_release",
+    "janus_collection_job_driver_release",
+    "janus_cli_release",
+  ]
+}
+
+group "interop_binaries_release" {
+  targets = [
+    "janus_interop_client_release",
+    "janus_interop_aggregator_release",
+    "janus_interop_collector_release",
+  ]
+}
+
+group "interop_binaries_small" {
+  targets = [
+    "janus_interop_client_small",
+    "janus_interop_aggregator_small",
+    "janus_interop_collector_small",
+  ]
+}
+
+target "janus_aggregator" {
+  args = {
+    GIT_REVISION = "${GIT_REVISION}"
+  }
+  cache-from = [
+    "type=gha,scope=main-janus",
```

Comment:
What happens when we `docker buildx bake` outside of GitHub Actions? Are the `cache-to/from` directives gracefully ignored?

Could we add a `CACHE_TYPE` variable that defaults to `local` so that we get a speedup when building images locally?

Reply:
If the `gha` cache backend is used outside of GitHub Actions, it fails silently, because it doesn't get either the cache API URL or the token from special environment variables. (This was a stumbling block for Jacob and I previously, but it also means these directives won't get in the way of local builds)

The documentation says that the "local BuildKit cache ... is always enabled", and empirically if I run `docker buildx bake` repeatedly, layer caching works as-is. I think the `local` cache backend is distinct, and writes unpacked OCI images to the filesystem at a user-defined location.